lib/errors/WWWAuthError.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) 2015-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+
+import type { HttpResponse } from '../http';
+import CustomError from './CustomError';
+import { isFunction } from '../util';
+
+// Error thrown after an unsuccessful network request which requires an Authorization header 
+// and returns a 4XX error with a www-authenticate header. The header value is parsed to construct 
+// an error instance, which contains key/value pairs parsed out
+export default class WWWAuthError extends CustomError {
```

Comment:
Cool! I wonder if there's a way to leverage this utility for step up auth error responses. Will have to experiment. I ended up having to write my own WWW-Authenticate header parser